Support for review requests (#576)
47degrees/github4s: microsite/docs/pull_request.md

See [the API doc](https://developer.github.com/v3/pulls/reviews/#create-a-review-for-a-pull-request) for full reference.

## Review requests

> This API allows you to operate review requests. Reviewers could be both users and teams. Usernames should be used without a leading '@' sign.

Reviewer: ```suggestion
This API allows you to operate on review requests. Reviewers can be users and/or teams. Usernames should be used without a leading '@' sign.
```

> **NOTE**: There might be problems with team reviewers and Github App authentication. 

Reviewer: can you expand a bit on that? what's the issue?

Author: Got errors trying to request a review from a team via github app auth. On a second thought, that most likely was a misconfiguration on the app's permissions, so I'd rather drop that note.

- the repository coordinates (`owner` and `name` of the repository).
- the pull request id.
- users and teams you want to remove from reviewers

As an example, if we wanted to remove `torvalds` from the reviewers for pull request 139 of `47degrees/github4s`:

```scala mdoc:compile-only
import github4s.domain._
val removeReviewers = gh.pullRequests.removeReviewers(
  "47deg",
  "github4s",
  139,
  RequestedReviewersRequest(List("torvalds")))
val response = removeReviewers.unsafeRunSync()
response.result match {
  case Left(e) => println(s"Something went wrong: ${e.getMessage}")
  case Right(r) => println(r)
}
```

The `result` on the right is the updated [PullRequest][pr-scala]. 

See [the API doc](https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#remove-requested-reviewers-from-a-pull-request) for full reference.

As you can see, a few features of the pull request endpoint are missing. As a result, if you'd like
to see a feature supported, feel free to create an issue and/or a pull request!

[pr-scala]: https://github.com/47degrees/github4s/blob/master/github4s/src/main/scala/github4s/domain/PullRequest.scala

> [rr-scala]: https://github.com/47degrees/github4s/blob/master/github4s/src/main/scala/github4s/domain/RequestedReviewersResponse.scala

Reviewer: I think you put the case classes in `PullRequest.scala`
